### PR TITLE
MM-3543/MM-35436: fix thread being marked unread when updating reply posts

### PR DESCRIPTION
**Pull request by ashishbhate**
#### Summary

- Threads should not be returned as unread if a reply post has been edited or reacted to since the time the thread was last viewed

#### Ticket Link

- https://mattermost.atlassian.net/browse/MM-35436
- https://mattermost.atlassian.net/browse/MM-35437
 
#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by mickmister — COMMENTED** [review]
LGTM :+1:  though I'd like to opt for someone else's review that is more familiar with the threaded reply functionality

File: api4/user_test.go [hunk 1]
```diff
@@ -5515,6 +5515,51 @@ func TestGetThreadsForUser(t *testing.T) {
 		require.NotNil(t, uss3.Threads)
 		require.Len(t, uss3.Threads, 0)
 	})
+
+	t.Run("editing or reacting to reply post does not make thread unread", func(t *testing.T) {
+		Client := th.Client
+
+		rootPost, _ := postAndCheck(t, Client, &model.Post{ChannelId: th.BasicChannel.Id, Message: "root post"})
```
Comment (isacikgoz) [inline api4/user_test.go:5522]: Shall we check the response errors?

**Review by isacikgoz — APPROVED** [review]
LGTM, just a general thing about error checking to avoid obscure fails.

**Comment by ashishbhate** [comment]
>LGTM 👍 though I'd like to opt for someone else's review that is more familiar with the threaded reply functionality

Currently very few devs are familiar with the threaded reply functionality. I'm trying to increase the bus factor by asking devs from the messaging team to review smaller threads PRs.

File: api4/user_test.go (hunk 1 above)
Comment (ashishbhate) [reply api4/user_test.go:5522]: The `postAndCheck` function checks the response for errors. I'm not sure why `postAndCheck` also returns the response, it isn't used anywhere.

**Review by ogi-m — APPROVED** [review]
LGTM. 

Editing a reply post still updates the timestamp, but I will open a separate ticket for that.

**Pull request merged**